lib/cocoapods-core/specification/linter.rb

```diff
@@ -387,8 +388,15 @@ def perform_github_uri_checks(git, git_uri)
       # Performs validations related to SSH sources
       #
       def check_git_ssh_source(s)
+        require 'uri'
+
         if git = s[:git]
-          if git =~ /\w+\@(\w|\.)+\:(\/\w+)*/
+          if git =~ /^#{URI.regexp}$/
```

Comment:
The essential change of the pull request is not this.
The change is:
`/\w+\@(\w|\.)+\:(\/\w+)*/` on line 391 to
`/(\w+\@)?(\w|\.)+\:(\/\w+)*/` on line 398.

I hope this handle `gitbucket:/group/repo.git` as a Git SSH source, same as `git@bitbucket.org:kylef/test.git`.
The difference is whether the source contains `user@` before its hostname.
Both cases are now in `linter_spec.rb`. `it 'warns for SSH repositories'` and `it 'warns for SSH repositories (without user@)'`.

Above block includes `/^#{URI.regexp}$/` is a guard introduced in this PR in order to eliminate non-SSH urls, because `check_git_ssh_source` can be called with a non-SSH url.